internal/resources/resource_host.go

```diff
@@ -481,6 +488,14 @@ func resourceMetalHostRead(d *schema.ResourceData, meta interface{}) (err error)
 			vTargetIQN:   i.TargetIQN,
 		}
 		volumeInfos = append(volumeInfos, vi)
+
+		if discoveryIP == "" {
+			discoveryIP = i.DiscoveryIP
+		}
+	}
+
+	if err = d.Set(hISCSIDiscoveryAddress, discoveryIP); err != nil {
+		return fmt.Errorf("set discovery IP: %v", err)
 	}
```

Comment:
why is this needed when discovery IP is already in volumeInfos?

Reply:
Good point, either application has to loop through Volume Info or we have to...... Let me ask the users.

Reply:
if volume info slice is not empty, they would get the first entry to get the discoveryIP right?

Reply:
Actually, in this case which the host is there. In theory, they know which VolumeInfo to use to fetch the DiscoveryIP. I think pushing back to them is a better solution. Thanks